File: Easy/Javascript/234. Palindrome Linked List.js
/**Given the head of a singly linked list, return true if it is a 
palindrome or false otherwise. */


/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */

 
/**
 * @param {ListNode} head
 * @return {boolean}
 */
var isPalindrome = function(head) {
    let arr = [];
    let curr = head;

    while(curr){
        arr.push(curr.val);
        curr = curr.next;
    }

    return arr.join('') === arr.reverse().join('')
};

File: Easy/Python/234. Palindrome Linked List.py
'''
Given the head of a singly linked list, return true if it is a 
palindrome or false otherwise.
'''

class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def isPalindrome(self, head: ListNode) -> bool:
        palindrome = []
        curr = head

        while curr:
            palindrome.append(curr.val)
            curr = curr.next

        return palindrome == palindrome[::-1]